### PR TITLE
Adding graphql functionality to quickstart

### LINKED ISSUE #150: Add a GraphQL API to the application
## Type of Issue:

- [x] Enhancement: New feature or request

#### Issue Creator Checklist

- [x] This issue has been thoroughly documented below; a developer should be able to understand the issue by reading it.

## Background

The MDCT team would like to start using GraphQL as part of the applications interaction with the Dynamo DB.  GraphQL offers the team a standard more "relational" syntax in interfacing with the DB (Its a specific API that does something very well and is becoming more widely adopted).  The purpose of this ticket is to incorporate graphql functionality into the quickstart app.

## Proposed Approach

Appsync is typically how we would resolve this issue. However, Appsync is not fedramp approved nor cms cloud approved. The time to get that approved and overhead involved is not feasible given the desired turnaround. 

The approach used elsewhere (specifically by the mc-review team) is to use Apollo server library running in lambda behind an App API and Apollo client library in the react code to interact with Apollo. 

Security for the App API we believe can be provided by the existing cognito user pool.

## Value

A standardized API call to the Dynamo DB for reading, writing, querying, etc.

## AC:
- Theres an App API for GraphQL 
- Lambdas will be created behind the API that utilize graphql to query the existing Dynamo
- The service will be protected by Cognito authentication 
- The GraphQL workflow will be utilized within the App itself (to be updated)
- Local workflow should be supported
- Update the quickstart environment diagram

### PR DESCRIPTION
**Pull request by joeaero19**
## Purpose

The purpose of this ticket is to incorporate graphql functionality into the quickstart app.

#### Linked Issues to Close
Closes #150 

## Approach

_How does this change address the issue?_
The approach used is to use Apollo server library running in lambda behind an App API and Apollo client library in the react code to interact with Apollo.

App sync would have been an alternative solution but it is not fedramp approved.

## Learning
There is example in the graphql documentation on how to integrate this.
https://www.apollographql.com/docs/react/get-started/

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -69,15 +69,15 @@ jobs:
       - name: set path
         run: |
           echo "PATH=$(pwd)/node_modules/.bin/:$PATH" >> $GITHUB_ENV
-      - name: unit test & publish coverage
```
Comment (dwhitecl) [inline .github/workflows/deploy.yml:72]: why did you comment this out?

File: services/ui-src/.env [hunk 2]
```diff
@@ -1,11 +1,12 @@
 LOCAL_LOGIN=true
 API_REGION=us-east-1
 API_URL=https://c1ng7kke96.execute-api.us-east-1.amazonaws.com/dev
+API_GRAPHQL_REGION=us-east-1
+API_GRAPHQL_URL=place_holder
```
Comment (dwhitecl) [inline services/ui-src/.env:5]: ?

Comment (joeaero19) [reply services/ui-src/.env:5]: This is a placeholder for the graphql environment values. These values will be populated once Api url is generated.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (joeaero19) [reply .github/workflows/deploy.yml]: I commented it out because it was failing before. I uncommented it . it looks good now.

**Pull request closed without merging**